internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -243,6 +255,37 @@ func (r *BMACReconciler) reconcileAgentSpec(bmh *bmh_v1alpha1.BareMetalHost, age
 	return reconcileComplete{true}
 }
 
+// The detached annotation is added if the installation of the agent associated with
+// the host has started.
+func (r *BMACReconciler) addBMHDetachedAnnotationIfAgentHasStartedInstallation(ctx context.Context, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
+
+	bmhAnnotations := bmh.ObjectMeta.GetAnnotations()
+	// Annotation already set to true
+	if bmhAnnotations[BMH_DETACHED_ANNOTATION] == "true" {
+		return reconcileComplete{}
+	}
```

Comment:
What do you thikn about doing something like what BMO does? 

https://github.com/metal3-io/baremetal-operator/blob/a39c5d3866ee3554417e11075f535b13275ff396/controllers/metal3.io/host_state_machine.go#L245-L247

It'd be better to not check for the value, but rather just the existence of the annotation

Reply:
Yes, that is more consistent with BMO. Will check for existence.